keras_cv/layers/object_detection/iou_similarity.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+class IouSimilarity(tf.keras.layers.Layer):
+    """Computes a lookup table matrix containing the ious for a given pair of boxes.
```

Comment:
Maybe /s/lookup table matrix/similarity matrix

I think it would be good if the docstring refers to what's returned with the same phrase throughout (currently it calls it both a lookup table matrix and a lookup vector and an iou_lookup_table